Tests/Unit/Consumer/AbstractAsyncConsumerTest.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Smartbox\Integration\FrameworkBundle\Tests\Unit\Consumer;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\AbstractAsyncConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
+use Smartbox\Integration\FrameworkBundle\Events\TimingEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * Class AbstractAsyncConsumerTest
+ * @package Smartbox\Integration\FrameworkBundle\Tests\Unit\Consumer
+ *
+ * @group abstract-async-consumer
+ * @group time-sensitive
+ */
+class AbstractAsyncConsumerTest extends TestCase
```

Comment:
I will finish the review of this class once the comments of method visibility in`AbstractAsyncConsumer`  has been discussed

Reply:
Alright.

Reply:
Test reviewed